mission/gui_auv/auv_gui/auv_gui.py

```diff
@@ -0,0 +1,283 @@
+import math
+import sys
+from threading import Thread
+from typing import List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rclpy
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
+from nav_msgs.msg import Odometry
+from PyQt5.QtCore import QTimer
+from PyQt5.QtWidgets import (
+    QApplication,
+    QGridLayout,
+    QLabel,
+    QMainWindow,
+    QTabWidget,
+    QVBoxLayout,
+    QWidget,
+)
```

Comment:
Consider switching to PyQt6. The differences between 5 and 6 are few, but would still recommend 6